src/blueprint/from_config/operators/ref_field.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::BTreeMap;
+
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, Field};
+use crate::lambda::Expression;
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+fn fields_has_resolver(ty: &config::Type, types: &BTreeMap<String, config::Type>) -> bool {
```

Comment:
isn't this almost the save function as `validate_field_of_type_has_resolver` below? at least they have the same purpose to check nested resolvers.

So, I'm thinking since `update_ref_field` returns `Valid` we can have that logic for validation and setting resolver completely in it and there is no need to do the same inside `src/blueprint/from_config/schema.rs`

Reply:
we don't have specific separation for updating and validation as it usually easier to make both together since they usually involve the same kind of operations to do. Take a look at other fields like `src/blueprint/from_config/operators/modify.rs` to confirm this.